InfiniTAM/ITMLib/Engine/ITMExternalTracker.h

```diff
@@ -0,0 +1,104 @@
+// Copyright 2014-2015 Isis Innovation Limited and the authors of InfiniTAM
+
+#pragma once
+
+#include <ros/ros.h>
+#include <tf/transform_listener.h>
+
+#include "../Utils/ITMLibDefines.h"
+
+#include "../Objects/ITMImageHierarchy.h"
+#include "../Objects/ITMViewHierarchyLevel.h"
+
+#include "../Engine/ITMLowLevelEngine.h"
+#include "../Engine/ITMTracker.h"
+
+using namespace ITMLib::Objects;
+
+namespace ITMLib {
+namespace Engine {
+/** Base class for engines using an external tracking system (e.g. Vicon).
+   Implementations would typically take the odometry or pose measurements from
+   the external tracking system and get the current view frame from it.
+*/
+class ITMExternalTracker : public ITMTracker {
+ private:
+  const ITMLowLevelEngine* lowLevelEngine;
+
+  void PrepareForEvaluation(const ITMView* view);
+
+ protected:
+  TrackerIterationType iterationType;
```

Comment:
Those variable names were already here, it appears that the authors did not use the same rules all the time. Sometimes they use camelCase, sometimes CamelCase and sometimes camel_case ;)